webdriver/tests/bidi/browsing_context/set_viewport/set_viewport.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+
+from ... import get_viewport_dimensions
+
+
+@pytest.mark.asyncio
+async def test_set_viewport(bidi_session, new_tab):
+    test_viewport = {"width": 250, "height": 300}
+
+    await bidi_session.browsing_context.set_viewport(
+        context=new_tab["context"],
+        viewport=test_viewport)
+
+    assert await get_viewport_dimensions(bidi_session, new_tab) == test_viewport
+
+
+@pytest.mark.asyncio
+async def test_set_viewport_reset(bidi_session, new_tab):
+    original_viewport = await get_viewport_dimensions(bidi_session, new_tab)
+
+    test_viewport = {"width": 666, "height": 333}
+    await bidi_session.browsing_context.set_viewport(
+      context=new_tab["context"],
+      viewport=test_viewport)
+
+    assert await get_viewport_dimensions(bidi_session, new_tab) == test_viewport
+
+    await bidi_session.browsing_context.set_viewport(
+      context=new_tab["context"],
+      viewport=None)
+    assert await get_viewport_dimensions(bidi_session, new_tab) == original_viewport
+
+
+@pytest.mark.asyncio
+async def test_set_viewport_affects_specific_context(bidi_session, top_context, new_tab):
+    original_viewport = await get_viewport_dimensions(bidi_session, top_context)
+
+    test_viewport = {"width": 333, "height": 333}
+    await bidi_session.browsing_context.set_viewport(
+      context=new_tab["context"],
+      viewport=test_viewport)
+
+    assert await get_viewport_dimensions(bidi_session, new_tab) == test_viewport
+    assert await get_viewport_dimensions(bidi_session, top_context) == original_viewport
+
+
+@pytest.mark.parametrize("protocol,parameters", [
+    ("http", ""),
+    ("https", ""),
+    ("https", {"pipe": "header(Cross-Origin-Opener-Policy,same-origin)"})
+], ids=[
+    "http",
+    "https",
+    "https coop"
+])
+@pytest.mark.asyncio
+async def test_set_viewport_persists_on_navigation(bidi_session, new_tab, inline, protocol, parameters):
+    test_viewport = {"width": 499, "height": 599}
+
+    await bidi_session.browsing_context.set_viewport(
+        context=new_tab["context"],
+        viewport=test_viewport)
+
+    assert await get_viewport_dimensions(bidi_session, new_tab) == test_viewport
+
+    url = inline("<div>foo</div>", parameters=parameters, protocol=protocol, domain="alt")
+    await bidi_session.browsing_context.navigate(
```

Comment:
We actually need two navigations to test this scenario. First navigate to the page without using `domain`, and setup the viewport. Then navigate to the alternative domain to trigger a COOP navigation. Right now we navigate from `about:blank` which might not trigger it.

Reply:
Updated

Reply:
It seems that it hasn't pushed yet?

Reply:
Should be pushed now